packages/sol-cov/src/trace.ts

```diff
@@ -0,0 +1,92 @@
+import { OpCode, StructLog, TransactionTrace } from '@0xproject/types';
+import { addressUtils, BigNumber, logUtils } from '@0xproject/utils';
+import { addHexPrefix, stripHexPrefix } from 'ethereumjs-util';
+import * as fs from 'fs';
+import * as _ from 'lodash';
+
+export interface TraceByContractAddress {
+    [contractAddress: string]: StructLog[];
+}
+
+export function getTracesByContractAddress(structLogs: StructLog[], startAddress: string): TraceByContractAddress {
+    const traceByContractAddress: TraceByContractAddress = {};
+    let currentTraceSegment = [];
+    const callStack = [startAddress];
+    // tslint:disable-next-line:prefer-for-of
+    for (let i = 0; i < structLogs.length; i++) {
+        const structLog = structLogs[i];
+        if (structLog.depth !== callStack.length - 1) {
+            throw new Error("Malformed trace. trace depth doesn't match call stack depth");
+        }
+        // After that check we have a guarantee that call stack is never empty
+        // If it would: callStack.length - 1 === structLog.depth === -1
+        // That means that we can always safely pop from it
+        currentTraceSegment.push(structLog);
+
+        if (_.includes([OpCode.CallCode, OpCode.StaticCall, OpCode.Call, OpCode.DelegateCall], structLog.op)) {
+            const currentAddress = _.last(callStack) as string;
+            const jumpAddressOffset = 1;
+            const newAddress = addressUtils.padZeros(
+                new BigNumber(addHexPrefix(structLog.stack[structLog.stack.length - jumpAddressOffset - 1])).toString(
```

Comment:
Yikes... This is a lot. Can we move this into a helper method? This is lower-level the the surrounding code. It'll be cleaner to have a well named method here.